collectors/o365/o365_collector.js

```diff
@@ -168,7 +169,20 @@ class O365Collector extends PawsCollector {
 
             return callback(null, logs, newState, newState.poll_interval_sec);
         }).catch(err => {
-            if (err.message) {
+            // set errorCode to showcase client error on DDMetric
+            if (typeof err === 'object') {
+                err.errorCode = err.code ? err.code : (err.statusCode ? err.statusCode : err.status);
+            }
+            // Handle the MS management api maximum palyload issue
+            if (err.message && err.message.indexOf('Maximum payload size exceeded') !== -1) {
```

Comment:
can we reply on some `statusCode` or similar instead of comparing strings?

Reply:
Actually i don't get complete error object. We only saw error message on cloud watch and it also not reproduce on local to get error object; so i am checking from message.